### PR TITLE
New package: hctl-0.2.6

### PR DESCRIPTION
**Pull request by michalszmidt**
#### Testing the changes
- I tested the changes in this PR: **YES**

_note:_
- `xlint` reports no problems
- tested with `XBPS_CHECK_PKGS=yes` and `-Q`

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [x] aarch64-glibc
  - [x] armv7l-glibc
  - [x] armv6l-glibc
  - [x] i686-glibc

_note:_
- not tested on void `*-musl` **but**, on alpine succeeds with all arches including riscv64

_About:_
This tool allows to manage host domain blocking (hostlist generation from multiple sources, record de-duplication). It operates in parallel. Also it's optimized for memory. It's cool comparing to some nasty scripts or sed especially when you count records in millions.

_Info:_

[Packaging status](https://repology.org/project/hctl/versions)

[Alpine Linux Package request](https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/51062) - uses musl, may be helpful for bug tracing, _accepted to testing_

_Disclaimer:_
I'm a hctl author.

Have a nice day!

### CODE REVIEW

**Comment by classabbyamp** [comment]
> Do you need Cargo.lock, or this approach is acceptable?

it is [recommended rust crate practice](https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html) to track cargo.lock in git for applications.

Also, we prefer that package template maintainers are not the upstream maintainers unless they use Void.

**Comment by michalszmidt** [comment]
Ok, so required changes are: 
- add Cargo.toml now as a separate file. (With next stable release I'll include it upstream)
- last line of template should not be empty (my nasty editor had to add it on save)

_I'll amend and fore push_

Questions:
- Do you wish to change commit message to some specific?
- Alpine uses cargo-auditable, should it be used here?

Void glibc is my default desktop OS, but you decide.

File: srcpkgs/hctl/template [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'hctl'
+pkgname=hctl
+version=0.2.2
+revision=1
+archs="i686* x86_64* aarch64* armv6* armv7*"
```
Comment (classabbyamp) [inline srcpkgs/hctl/template:5]:
```suggestion
```
no `archs=` is equivalent with "all archs"

File: srcpkgs/hctl/template [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+build_style=cargo
+hostmakedepends="pkg-config base-devel"
```
Comment (classabbyamp) [inline srcpkgs/hctl/template:7]:
```suggestion
hostmakedepends="pkg-config"
```

File: srcpkgs/hctl/template [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+depends="openssl"
+short_desc="CLI tool to merge hostlists for dns blocker like pihole"
+maintainer="Michał Szmidt <michal.sz@disroot.org>"
+license="BSD-3-Clause-No-Military-License"
+homepage="https://github.com/michalszmidt/hctl"
+distfiles="https://github.com/michalszmidt/hctl/archive/refs/tags/${version}.tar.gz"
+checksum=8b74e520a8412d39b1abe6ac944dc2e2c454c66affb227569f7d363f14deb5d3
+
+pre_build() {
+	cargo update
```
Comment (classabbyamp) [inline srcpkgs/hctl/template:18]: > add Cargo.[lock] now as a separate file. (With next stable release I'll include it upstream)

for now, it's better to just use `cargo generate-lockfile` for this

File: srcpkgs/hctl/template [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+post_install() {
+	vlicense LICENSE
+}
+
```
Comment (classabbyamp) [inline srcpkgs/hctl/template:24]:
```suggestion
```

**Review by classabbyamp — COMMENTED** [review]
> Do you wish to change commit message to some specific?

see [CONTRIBUTING.md](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#committing-your-changes)

> Alpine uses cargo-auditable, should it be used here?

[it's already being used](https://github.com/void-linux/void-packages/blob/master/common/build-style/cargo.sh)

> Void glibc is my default desktop OS, but you decide.

ok that's fine, just checking. we just want to ensure that the template maintainer actually tests the software on void.

**Comment by michalszmidt** [comment]
Done:
- "see [CONTRIBUTING.md](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#committing-your-changes)"
- _no archs= is equivalent with "all archs"_
- _`hostmakedepends="pkg-config"`_
- _for now, it's better to just use `cargo generate-lockfile` for this_

File: srcpkgs/hctl/template [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'hctl'
+pkgname=hctl
+version=0.2.2
+revision=1
+build_style=cargo
+hostmakedepends="pkg-config"
+makedepends="openssl-devel"
+depends="openssl"
```
Comment (classabbyamp) [inline srcpkgs/hctl/template:8]: not needed, this should be automatically detected

Comment (michalszmidt) [reply srcpkgs/hctl/template:8]: When removed `build_style=cargo` got:
```bash
=> ERROR: hctl-0.2.2_1: missing do_install() function!

```

When removed:
```ini
hostmakedepends="pkg-config"
makedepends="openssl-devel"
depends="openssl"
```
Got message:
`(...) The pkg-config command could not be found. (...)`

And when added pkg-config
`Package openssl was not found in the pkg-config search path.`

Are you sure?

Comment (classabbyamp) [reply srcpkgs/hctl/template:8]: I'm talking about depends=openssl

Comment (michalszmidt) [reply srcpkgs/hctl/template]: Right, it passes without `depends=openssl`

One more thing. Alpine Maintainers [protested against](https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/51062#note_335395) the [BSD-3-Clause-**No-Military**](https://spdx.org/licenses/BSD-3-Clause-No-Military-License.html) so I must have changed it to just [BSD-3-Clause](https://spdx.org/licenses/BSD-3-Clause.html). I rebased amended and force pushed [code for 0.2.2 commit](https://github.com/michalszmidt/hctl/commit/7897bd115bf279c08b0971509006afc375dd6fae) adding new license there but old archive is glued to release. So i made a new tag 0.2.2.1 with modified license.

So next force push contains also License change and checksum update.
Also `revision=0`as this new package.

**Comment by michalszmidt** [comment]
@classabbyamp 
Is there anything else I need to do?

**Comment by michalszmidt** [comment]
Updated to 0.2.4, as promised now Cargo.lock is upstream

**Comment by michalszmidt** [comment]
@classabbyamp 
Updated to **0.2.5** and rebased, is there anything else I can do?

**Review by melroy89 — APPROVED** [review]

**Comment by melroy89** [comment]
LGTM now..?

**Comment by michalszmidt** [comment]
> LGTM now..?

Update to 0.2.6

- Semi-important dependencies update
- New features
- Requires rust 1.74+

Also, all desired features are highlighted, published a roadmap to 1.0.0

**Comment by melroy89** [comment]
BUMP! And merge plz

**Pull request closed without merging**